openwisp_firmware_upgrader/apps.py

```diff
@@ -33,3 +36,12 @@ def add_default_menu_items(self):
         else:
             current_menu = getattr(settings, menu_setting)
             current_menu += items
+
+    def connect_device_signals(self):
+        DeviceConnection = load_model('connection', 'DeviceConnection')
+        DeviceFirmware = load_model('firmware_upgrader', 'DeviceFirmware')
+        post_save.connect(
+            DeviceFirmware.auto_add_device_firmware_to_device,
+            sender=DeviceConnection,
+            dispatch_uid='connection.auto_add_device_firmware',
+        )
```

Comment:
I thought about another use case that I'm sure it will happen for reasons I won't explain here for brevity.

It will happen that default credentials with `auto_add` are already defined, hence a `DeviceConnection` will be created before firmware images are uploaded into the syste, which will cause devices to be orphans of `DeviceFirmware` again.

To avoid this it should be enough to add another receiver to the `post_save` of `FirmwareImage`, something like this:
```python
def receiver_xyz(instance, created):
    if created:
        transaction.on_commit(lambda: create_all_device_firmwares.delay(instance.pk))
```

The task would be something like:
```python
@shared_task
create_all_device_firmwares(firmware_image_pk):
    queryset = Device.objects.filter(os=fw_image.build.os)
    if not queryset.exists():
        return
    fw_image = FirmwareImage.objects.get(pk=firmware_image_pk).select_related('build')
    for device in queryset:
        DeviceFirmware.create_for_device(device, firmware_image)
```

In this example I added the ``firmware_image`` argument, since we know it beforehand it would be good to avoid querying the DB again.